parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -488,31 +488,39 @@ loop = do
           ppe <- PPE.suffixifiedPPE <$> prettyPrintEnvDecl (Names names0 mempty)
           case (srcle, srclt, mdValuel) of
             (srcle, srclt, [r@(Referent.Ref mdValue)]) -> do
-              mdType <- eval $ LoadTypeOfTerm mdValue
-              case mdType of
-                Nothing -> respond $ MetadataMissingType ppe r
-                Just ty -> do
-                  let steps =
-                        second (const . step $ Type.toReference ty)
-                        . first (Path.unabsolute .  resolveToAbsolute) <$> srcs
-                  let get = Branch.head <$> use root
-                  before <- get
-                  stepManyAt steps
-                  after <- get
-                  (ppe, outputDiff) <- diffHelper before after
-                  respondNumbered $ ShowDiffNamespace
-                     Path.absoluteEmpty Path.absoluteEmpty ppe outputDiff
-                where
-                step mdType b0 = let
-                  tmUpdates terms = foldl' go terms srcle
-                    where
-                    go terms src = op (src, mdType, mdValue) terms
-                  tyUpdates types = foldl' go types srclt
-                    where
-                    go types src = op (src, mdType, mdValue) types
-                  in over Branch.terms tmUpdates . over Branch.types tyUpdates $ b0
-            (_srcle, _srclt, mdValues) ->
-              respond $ MetadataAmbiguous ppe mdValues
+             let defnsAndTerms = Map.fromList $ srcs `zip` (toList . getHQ'Terms <$> srcs)
+             let noReferents = Map.keys $ Map.filterWithKey (\_ v -> null $ v) defnsAndTerms
+             let paths = (\(path', segment) -> Path.snoc' path' $ HQ'.toName segment) <$> noReferents
+             when (not . null $ noReferents) $ respond $ DefinitionsNotFound paths
```

Comment:
Should this reuse `SearchTermsNotFound` instead of introducing a new `Output` type?

Reply:
Should be doable indeed, I think I went down this path cause I could not find a way to go from `(Path', NameSegment.HQSegment)` or from `Path'` to `HashQualified` which is what `SearchTermsNotFound` needs. :thinking:

Reply:
So from what I can tell about HQ and HQ':
* `manageLinks` takes a `[(Path', HQ'.HashQualified' NameSegment)]`, where `HQ'.HashQualified` and NameSegment` is equivalent to `NameSegment.HQSegment`.

* I can change the `paths` val to:
```haskell
let paths :: [HQ.HashQualified' Path'] = HQ'.toHQ <$> Path.unsplitHQ' <$> noReferents
```
meaning I am how in HQ land, but still there appears to be no way to convert from `HQ.HashQualified' Path'` to `HQ.HashQualified' Name`, which is what `SearchTermsNotFound` needs. 

Now at the same time, I am of the impression that an equivalent restriction exists for other error messages:

```haskell
  | PatchNotFound Path.Split'
  | TypeNotFound Path.HQSplit'
  | TermNotFound Path.HQSplit'
  | TypeNotFound' ShortHash
  | TermNotFound' ShortHash
  | SearchTermsNotFound [HQ.HashQualified] -- HQ.HashQualified' Name
```
I think that in this instance since the `Path.HQSplit' or `Path', HQ'.HashQualified' NameSegment` is indeed what the user has requested gets linked, it is what should be serialised in the response.

Reply:
Updated to reuse `SearchTermsNotFound`. As discussed, with an intermediate cast so that the `Text` is extracted from the `Path'` and used as a `NameOnly Name`